app/controllers/translations_controller.rb

```
@@ -1,24 +1,33 @@
 class TranslationsController < ApplicationController
   # GET /translations
   # GET /translations.json
-
   before_filter :authenticate_translator!
+  load_and_authorize_resource
 
   def index
-    @locale = Locale.find(params[:locale_id])
-    @translations = @locale.translations.all
-
-    respond_to do |format|
-      format.html # index.html.erb
-      format.json { render json: @translations }
+    if !Locale.find(params[:locale_id]).nil?
```

should raise an error if no locale found.
i.e. ActiveRecord::RecordNotFound

Use Locale.find_by_id(params[:locale_id]) instead

@locale = Locale.find_by_id(params[:locale_id])
if @locale 
  @translations = @locale.translations (no need for the .all)
else
  @translations = Translation.all
end